admin/data/sql/migrations/153.do.fix-spCreateChecks.sql

```diff
@@ -0,0 +1,81 @@
+
+ALTER PROCEDURE [mtc_admin].[spCreateChecks]
+    @TVP [mtc_admin].[CheckTableType] READONLY
+AS
+
+  -- Connection pooling is enabled - make sure we are not in an existing transaction
+  IF @@TRANCOUNT <> 0  -- Rollback old transactions before starting another
+    ROLLBACK TRANSACTION
+
+  BEGIN TRY
+  BEGIN TRANSACTION
+
+  DECLARE checkArgsList CURSOR
+  FOR SELECT pupil_id, checkForm_id, checkWindow_id, isLiveCheck, pinExpiresAt, school_id
+      FROM @TVP
+    FOR READ ONLY
+
+  DECLARE @pupilId int
+  DECLARE @checkFormId int
+  DECLARE @checkWindowId int
+  DECLARE @isLiveCheck bit
+  DECLARE @pinExpiresAt datetimeoffset
+  DECLARE @schoolId int
+  DEClARE @checkId int
+  DECLARE @output TABLE (id int);
+
+  OPEN checkArgsList
+  FETCH checkArgsList INTO @pupilId, @checkFormId, @checkWindowId, @isLiveCheck, @pinExpiresAt, @schoolId
+  WHILE (@@FETCH_STATUS = 0) BEGIN
+
+    -- Create the check
+    INSERT INTO [mtc_admin].[check]
+        (pupil_id, checkForm_id, checkWindow_id, isLiveCheck)
+    VALUES (@pupilId, @checkFormId, @checkWindowId, @isLiveCheck);
+
+    -- Get the check.id we just inserted
+    SET @checkId = SCOPE_IDENTITY();
+
+    -- Assign a pin to the check
+    INSERT INTO [mtc_admin].[checkPin] (school_id, check_id, pinExpiresAt, pin_id)
+    VALUES (@schoolId, @checkId, @pinExpiresAt, (SELECT TOP 1 p.id FROM mtc_admin.pin p 
```

Comment:
@GuyHarwood minor comment: can you split that sql in different lines to make it more readable please?